semcore/outside-click/src/OutsideClick.tsx

```diff
@@ -36,6 +39,11 @@ function OutsideClick(props: IFunctionProps<IOutsideClickProps>) {
 
   const handleRef = useForkRef(children ? children.ref : null, nodeRef, forwardRef!);
 
+  const isModalRoot = React.useCallback(
+    (root: HTMLElement) => root && root.dataset.uiName === 'Modal.Overlay',
```

Comment:
You don't need `root &`, because of types.

And I'm not sure about `useCallback`, it looks as just a utility function.

Maybe it is better to create function `toggleEvents` where you will check this and enable or disable events by some parameter?

Reply:
Rewritten to `toggleEvents` function remove code duplication.